controllers/cleaner_test.go

```diff
@@ -0,0 +1,37 @@
+package controllers
+
+import (
+	"sync"
+
+	"github.com/onsi/gomega"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type resourceToBeDeleted struct {
+	Namespace string
+	Name      string
+	Object    client.Object
+}
+
+func waitResourceToBeDelete(g gomega.Gomega, resource resourceToBeDeleted) {
```

Comment:
Can you just get the key from the object?

Reply:
I was thinking about it, I started to build a Cleaner and just call `cleaner.AddResource(object)` and a single defer on `cleaner.Cleanup()`. It had some issues because with that I moved the `k8sclient.Delete` call too, but it did not like some of the resources on the eventual part (never failed). I'm still trying to do that, but I wanted to put this out to run ci tests.

Reply:
I mean can you just pass `client.Object` here (and `[]client.Object` below), since the other fields in `resourceToBeDeleted` can be derived from the `client.Object`.

Reply:
Yes, that's exactly what I try to do, but somehow the `k8sClient.Get` call doesn't like that. That's why I'm still trying to do that and not pushed yet.